### PR TITLE
NAS-107445 / 12.0 / Remove unprotected setfacl / chmod call

### PR DESCRIPTION
**Pull request by anodos325**
Permissions operations during home directory move bypass our middleware API for
permissions. Move the copy and permissions operation into a middleware job and
use API.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/account.py [hunk 1]
```diff
@@ -692,6 +685,22 @@ async def set_root_password(self, app, password, options):
         root = await self.middleware.call('user.query', [('username', '=', 'root')], {'get': True})
         await self.middleware.call('user.update', root['id'], {'password': password})
 
+    @job(lock="home_copy")
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/account.py:688]: We can use lock that includes `home_new` in its name so multiple independent calls could be performed simultaneously

**Review by themylogin — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/account.py (hunk 1 above)
Comment (anodos325) [reply src/middlewared/middlewared/plugins/account.py:688]: Good idea

Comment (william-gr) [reply src/middlewared/middlewared/plugins/account.py:688]: Should probably mark it as private as well

Also we should probably keep the change as minimal as possible in the 12.0 backport

**Review by themylogin — APPROVED** [review]

**Review by william-gr — CHANGES_REQUESTED** [review]
Need to fix PEP8 before I can merge

**Review by william-gr — APPROVED** [review]

**Pull request merged**